src/mixtures/mixturemodel.jl

```diff
@@ -477,6 +477,22 @@ rand(rng::AbstractRNG, s::MixtureSampler{Univariate}) =
 rand(rng::AbstractRNG, d::MixtureModel{Univariate}) =
     rand(rng, component(d, rand(rng, d.prior)))
 
+function rand(rng::AbstractRNG, d::MixtureModel{Univariate}, n::Int)
+    counts = rand(rng, Multinomial(n, probs(d.prior)))
+    x = Vector{eltype(d)}(undef, n)
+    offset = 0
+    for i in eachindex(counts)
+        ni = counts[i]
+        if ni > 0
+            c = component(d, i)
+            v = view(x, (offset+1):(offset+ni))
+            v .= rand(rng, c, ni)
+            offset += ni
```

Comment:
Since `v` is updated in-place, we should avoid any additional allocations here and directly call
```suggestion
            last_offset = offset + ni - 1 
            rand!(rng, c, @view(x[(begin + offset):(begin + last_offset)]))
            offset = last_offset + 1
```